FreeRTOS-Plus/Demo/Device_Shadow_Windows_Simulator/Device_Shadow_Demo/DemoTasks/shadow_demo_helpers.c

```diff
@@ -0,0 +1,932 @@
+ *
+ * @return The status of the final connection attempt.
+ */
+static TlsTransportStatus_t prvConnectToServerWithBackoffRetries( NetworkContext_t * pxNetworkContext );
+
+/**
+ * @brief Function to get the free index at which an outgoing publish
+ * can be stored.
+ *
+ * @param[out] pucIndex The output parameter to return the index at which an
+ * outgoing publish message can be stored.
+ *
+ * @return EXIT_FAILURE if no more publishes can be stored;
+ * EXIT_SUCCESS if an index to store the next outgoing publish is obtained.
+ */
+static int prvGetNextFreeIndexForOutgoingPublishes( uint8_t * pucIndex );
+
+/**
+ * @brief Function to clean up an outgoing publish at given index from the
+ * #outgoingPublishPackets array.
+ *
+ * @param[in] ucIndex The index at which a publish message has to be cleaned up.
+ */
+static void vCleanupOutgoingPublishAt( uint8_t ucIndex );
+
+/**
+ * @brief Function to clean up all the outgoing publishes maintained in the
+ * array.
+ */
+static void vCleanupOutgoingPublishes( void );
+
+/**
+ * @brief Function to clean up the publish packet with the given packet id.
+ *
+ * @param[in] usPacketId Packet identifier of the packet to be cleaned up from
+ * the array.
+ */
+static void vCleanupOutgoingPublishWithPacketID( uint16_t usPacketId );
+
+/**
+ * @brief Function to resend the publishes if a session is re-established with
+ * the broker. This function handles the resending of the QoS1 publish packets,
+ * which are maintained locally.
+ *
+ * @param[in] pxMqttContext MQTT context pointer.
+ */
+static int handlePublishResend( MQTTContext_t * pxMqttContext );
+
+/**
+ * @brief The timer query function provided to the MQTT context.
+ *
+ * @return Time in milliseconds.
+ */
+static uint32_t prvGetTimeMs( void );
+
+/*-----------------------------------------------------------*/
+
+static TlsTransportStatus_t prvConnectToServerWithBackoffRetries( NetworkContext_t * pxNetworkContext )
+{
+    TlsTransportStatus_t xNetworkStatus = TLS_TRANSPORT_SUCCESS;
+    RetryUtilsStatus_t xRetryUtilsStatus = RetryUtilsSuccess;
+    RetryUtilsParams_t xReconnectParams = { 0 };
+    NetworkCredentials_t xNetworkCredentials = { 0 };
+    
+    /* Set the credentials for establishing a TLS connection. */
+    xNetworkCredentials.pRootCa = ( const unsigned char * ) democonfigROOT_CA_PEM;
+    xNetworkCredentials.rootCaSize = sizeof( democonfigROOT_CA_PEM );
+    #ifdef democonfigCLIENT_CERTIFICATE_PEM
+        xNetworkCredentials.pClientCert = ( const unsigned char * ) democonfigCLIENT_CERTIFICATE_PEM;
+        xNetworkCredentials.clientCertSize = sizeof( democonfigCLIENT_CERTIFICATE_PEM );
+        xNetworkCredentials.pPrivateKey = ( const unsigned char * ) democonfigCLIENT_PRIVATE_KEY_PEM;
+        xNetworkCredentials.privateKeySize = sizeof( democonfigCLIENT_PRIVATE_KEY_PEM );
+    #endif
+    #ifdef democonfigUSE_AWS_IOT_CORE_BROKER
+        xNetworkCredentials.disableSni = pdFALSE;
```

Comment:
Please remove this line. We always want to use SNI when connecting to AWS IoT Core.